lms/djangoapps/learner_dashboard/programs.py

```diff
@@ -70,14 +74,93 @@ class ProgramDetailsFragmentView(EdxFragmentView):
     """
     Render the program details fragment.
     """
+    DEFAULT_ROLE = 'student'
+
+    @staticmethod
+    def get_program_discussion_configuration(program_uuid):
+        return ProgramDiscussionsConfiguration.objects.filter(
+            program_uuid=program_uuid
+        ).first()
+
+    @staticmethod
+    def _get_resource_link_id(program_uuid, request) -> str:
+        site = get_current_site(request)
+        return f'{site.domain}-{program_uuid}'
+
+    @staticmethod
+    def _get_result_sourcedid(context_id, resource_link_id, user_id) -> str:
+        return f'{context_id}:{resource_link_id}:{user_id}'
+
+    def _get_lti_embed_code(self, program_discussions_configuration, request) -> str:
+        """
+        Returns the LTI embed code for embedding in the program discussions tab
+        Args:
+            program_discussions_configuration (ProgramDiscussionsConfiguration): ProgramDiscussionsConfiguration object.
+            request (HttpRequest): Request object for view in which LTI will be embedded.
+        Returns:
+            HTML code to embed LTI in program page.
+        """
+        program_uuid = program_discussions_configuration.program_uuid
+        lti_consumer = program_discussions_configuration.lti_configuration.get_lti_consumer()
+        user_id = str(request.user.id)
+        context_id = program_uuid
+        resource_link_id = self._get_resource_link_id(program_uuid, request)
+        # TODO: Add support for multiple roles
+        roles = self.DEFAULT_ROLE
+        context_title = program_uuid
+        result_sourcedid = self._get_result_sourcedid(context_id, resource_link_id, user_id)
+
+        return lti_embed(
+            html_element_id='lti-tab-launcher',
+            lti_consumer=lti_consumer,
+            resource_link_id=resource_link_id,
+            user_id=user_id,
+            roles=roles,
+            context_id=context_id,
+            context_title=context_title,
+            context_label=context_id,
+            result_sourcedid=result_sourcedid
+        )
+
+    def render_discussions_fragment(self, program_uuid, request) -> dict:
+        """
+        Returns the program discussion fragment if program discussions configuration exists for a program uuid
+        """
+        if program_discussions_is_enabled():
```

Comment:
to avoid indentation you can use "return early" principle, that is 
```
return {
            'iframe': '',
            'enabled': False
        }
```

in case of waffle is not available or program doesn't have a discussion config.